msbuild/Xamarin.iOS.Tasks.Core/Tasks/DetectDebugNetworkConfigurationTaskBase.cs

```diff
@@ -48,11 +50,26 @@ public override bool Execute ()
 					hosts = DebuggerHosts.Split (new [] { ';' }, StringSplitOptions.RemoveEmptyEntries);
 
 				if (hosts == null || hosts.Length == 0) {
+					var properties = IPGlobalProperties.GetIPGlobalProperties ();
+					var hostName = properties.HostName;
+
 					try {
-						ips.AddRange (Dns.GetHostEntry (Dns.GetHostName ()).AddressList.Select ((v) => v.ToString ()));
+						var entry = Dns.GetHostEntry (hostName);
+
+						ips.AddRange (entry.AddressList.Select (v => v.ToString ()));
 					} catch {
-						Log.MTError (7001, "Could not resolve host IPs for WiFi debugger settings.");
-						return false;
+						using (var socket = new Socket (SocketType.Stream, ProtocolType.Tcp)) {
```

Comment:
I do not think using `TCP` is a good idea because you will hang for some time and you will timeout if the address is not resolved (imagine being behind a firewall). It might be better to use UDP and a datagram socket the address does not even need to exist this way so doing this instead should do the trick.

```csharp
using (var socket = new Socket (SocketType.Dgram, ProtocolType.Udp)) {
```

Reply:
by doing this even doing something like `socket.Connect ("1.2.3.4", 80);` would work